OnCall open source edits (#82)
grafana/oncall: docs/sources/open-source.md

weight: 300
---

> # Grafana OnCall Open Source guide

Reviewer: ```suggestion
# Grafana OnCall open source guide
```

Grafana OnCall is a developer-friendly incident response tool that's available to Grafana open source and Grafana Cloud users. The OSS version of Grafana OnCall provides the same reliable on-call management solution along with the flexibility of a self-managed environment. 

This guide describes the necessary installation and configuration steps needed to configure OSS Grafana OnCall. 

The intended audience for this guide includes:
- Grafana open source admins who are responsible for deploying and configuring Grafana OnCall.
- Grafana open source users who need to configure SMS and phone notifications using Grafana Cloud.

## Install Grafana OnCall OSS

There are three Grafana OnCall OSS environments available:

- **Hobby** playground environment for local usage: [README.md](https://github.com/grafana/oncall#getting-started)
- **Development** environment for contributors: [DEVELOPER.md](https://github.com/grafana/oncall/blob/dev/DEVELOPER.md)

> - **Production** environment for reliable cloud installation: [Production Environment](#production-environment)

Reviewer: ```suggestion
- **Production** environment for reliable Cloud installation: [Production Environment](#production-environment)
```
I don't know what "reliable" means. I think I know, but this doesn't really explain the difference between this all important environment and the others...also, the Cloud designation is confusing because we're talking about OSS.

For detailed installation instructions and additional resources, refer to the OSS Grafana OnCall [README.md](https://github.com/grafana/oncall#getting-started)

For more information on production environment installation, refer to the following OSS Grafana OnCall [production environment helm chart](https://github.com/grafana/oncall/helm)

> ## Slack integration for Grafana OnCall OSS

Reviewer: Note that you'll want to edit the text that links to this topic. It looks like it hasn't been changed in Install Grafana OnCall OSS

For more information on production environment installation, refer to the following OSS Grafana OnCall [production environment helm chart](https://github.com/grafana/oncall/helm)

## Slack integration for Grafana OnCall OSS

## Production Environment
The Slack integration for Grafana OnCall leverages Slack API features to provide a customizable and useful integration.

> We prepared the helm chart for production environment: https://github.com/grafana/oncall/helm

Reviewer: I know it's mentioned earlier in the doc, but it would probably be a good idea to keep the link out to the Helm chart.

Author: The link to the helm chart is in the section above - Install Grafana OnCall OSS

> - Create a new Slack App from Grafana OnCall app manifest.

Reviewer: We don't need any of these "requirements," since the explanation on how to do these thing is a part of the configuration task.

Author: Got it I'll remove these from both slack and telegram

### Configure Slack for Grafana OnCall OSS

> 1. Ensure your Grafana OnCall environment is up and running.
> 
> 2. Grafana OnCall must be accessible through HTTPS. For development purposes, it's recommend to use [localtunnel](https://github.com/localtunnel/localtunnel). For production purposes, consider establishing a proper web server with HTTPS termination. 
> For localtunnel, refer to the following configuration:

Reviewer: ```suggestion
1. Ensure your Grafana OnCall environment is up and running.

1. Grafana OnCall must be accessible through HTTPS. For development purposes,  use [localtunnel](https://github.com/localtunnel/localtunnel). For production purposes, consider establishing a proper web server with HTTPS termination. 
For localtunnel, refer to the following configuration:
```

Everything under this step needs to be indented by two tab spaces, beginning with "For localtunnel,..."

```bash
# Choose the unique prefix instead of pretty-turkey-83
# Localtunnel will generate an url, e.g. https://pretty-turkey-83.loca.lt
# it is referred as <ONCALL_ENGINE_PUBLIC_URL> below
lt --port 8080 -s pretty-turkey-83 --print-requests
```

> 3. If using localtunnel, open your external URL and click **Continue** to allow requests to bypass the warning page.
> 
> 4. [Create a Slack Workspace](https://slack.com/create) for development, or use your company workspace.

Reviewer: ```suggestion
1. If using localtunnel, open your external URL and click **Continue** to allow requests to bypass the warning page.

1. [Create a Slack Workspace](https://slack.com/create) for development, or use your company workspace.
```

> 5. Go to https://api.slack.com/apps and click **Create New App** 

Reviewer: ```suggestion
1. Go to https://api.slack.com/apps and click **Create an App** .
```

> 6. Select `From an app manifest` option and select your workspace.

Reviewer: ```suggestion
1. Select `From an app manifest` option and select your workspace.
```

> 7. Copy and paste the following block. Be sure to replace `YOUR_BOT_NAME` and `ONCALL_ENGINE_PUBLIC_URL` fields with the appropriate information. 

Reviewer: ```suggestion
1. Replace the text with the following YAML code block . Be sure to replace `<YOUR_BOT_NAME>` and `<ONCALL_ENGINE_PUBLIC_URL>` fields with the appropriate information. 
```

Reviewer: The original formatting for the user variables is both a best practice, and the actual format in the YAML code block.

Reviewer: I'm going through this actual process as I review these steps

    socket_mode_enabled: false
  ```

> 6. Set environment variables by navigating to your Grafana OnCall, then click **Env Variables** and set the following:

Reviewer: ```suggestion
1. Set environment variables by navigating to your Grafana OnCall, then click **Env Variables** and set the following:
```
Where is this? In ChatOps -> slack? It's not clear and these fields aren't available in Cloud OnCall

    ```
    SLACK_CLIENT_OAUTH_ID = Basic Information -> App Credentials -> Client ID
    SLACK_CLIENT_OAUTH_SECRET = Basic Information -> App Credentials -> Client Secret
    SLACK_SIGNING_SECRET = Basic Information -> App Credentials -> Signing Secret
    SLACK_INSTALL_RETURN_REDIRECT_HOST = << OnCall external URL >>
    ```

> 7. From OnCall, navigate to **ChatOps**, select Slack and click **Install Slack integration**.
> 
> 8. Configure additional Slack settings as needed and your integration is ready for use.
> 

Reviewer: ```suggestion
1. In OnCall, navigate to **ChatOps**, select Slack and click **Install Slack integration**.

1. Configure additional Slack settings.

```

## Telegram integration for Grafana OnCall OSS

The Telegram integration for Grafana OnCall is designed for collaborative team work and improved incident response.
### Before your begin

Review the following requirements to configure Telegram for Grafana OnCall OSS:
- Ensure your Grafana OnCall environment is running.
- Grafana OnCall must be externally available and provide HTTPS endpoint to establish subscription on Telegram events.
- Create new Telegram Group and private Telegram channel for alerts.

> 

Reviewer: not necessary

### Configure Telegram for Grafana OnCall OSS

> 1. Ensure your OnCall environment is up and running.
> 
> 2. Request [BotFather](https://t.me/BotFather) for a key, then add your key in `TELEGRAM_TOKEN` in your Grafana OnCall **Env Variables**.

Reviewer: ```suggestion
1. Ensure your OnCall environment is up and running.

1. Request [BotFather](https://t.me/BotFather) for a key, then add your key in `TELEGRAM_TOKEN` in your Grafana OnCall **Env Variables**.
```

> 3. Set `TELEGRAM_WEBHOOK_HOST` with your external URL for your Grafana OnCall. 

Reviewer: ```suggestion
1. Set `TELEGRAM_WEBHOOK_HOST` with your external URL for your Grafana OnCall. 
```

> 4. From the **ChatOps** tab in Grafana OnCall, click **Telegram** and your integration is now ready for use.

Reviewer: ```suggestion
1. From the **ChatOps** tab in Grafana OnCall, click **Telegram**.
```

## Connect Grafana Cloud to Grafana OnCall OSS

Open source Grafana OnCall can be connected to Grafana Cloud to configure a variety of notifications.

The benefits of connecting to Grafana Cloud include:
- Heartbeat notification
- SMS for user notifications
- Phone calls for user notifications.

> To connect to Grafana Cloud, refer to the **Cloud** page in your OSS Grafana OnCall instance. 

Reviewer: **Cloud** page? Is this page only available in OSS?

Author: that was my interpretation but not confirmed
